### ISSUE TITLE
v0.9 - API mode fails silently if setting a param that doesn't exist.

### ISSUE THREAD

**Issue by isaacovercast** [dereneaton/ipyrad#354]
Current behavior:
```
data = ipyrad.load_json("watdo.json")
data.params.min_samples_locs = 15
## fails silently
```

I would expect this to raise an error, otherwise this can lead to a condition where people think they are setting parameters and it just isn't doing it. Tricky to debug.

**Comment by dereneaton** [comment]
Ugh, I noticed that, I think we need to create or modify the `__settattr__` function of the Params class object to raise a key error if the key being set is not in `self._keys`. I started to mess with it once but didn't figure it out. 

TODO

**Comment by isaacovercast** [comment]
Fixed: d314684

This was __exactly__ as annoying to figure out as you might imagine.
